server/src/routes/tableaux/atlas/index.js

```diff
@@ -494,7 +494,13 @@ router.route('/atlas/get-filieres')
 
 router.route('/atlas/get-filters-values')
   .get(async (req, res) => {
-    const annee_universitaire = await db.collection('atlas2023').distinct("annee_universitaire");
+    let filters = {};
+    if (req.query.geo_id) {
+      filters = { geo_id: req.query.geo_id };
+    }
+
+    const annees_universitaires_onlyData = await db.collection('atlas2023').distinct("annee_universitaire", filters);
+    const annees_universitaires_all = await db.collection('atlas2023').distinct("annee_universitaire");
```

Comment:
Do you really need to query twice the database ? Would it be more efficient to post filter `annees_universitaires_all` in order to create `annees_universitaires_onlyData` ?

Reply:
the object returned will be very large. I think that's less efficient to set it in js code instead of disctinct in mongo. However, I'll be sure to check the indexes in database